pilot/cmd/pilot-agent/config_test.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2020 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/gogo/protobuf/types"
+
+	meshconfig "istio.io/api/mesh/v1alpha1"
+
+	"istio.io/istio/pkg/config/mesh"
+)
+
+func TestGetMeshConfig(t *testing.T) {
+	overrides := `
+defaultConfig:
```

Comment:
Just for my sanity, can we add a test case for a field with arrays and ensure that these are overrides/replacements and not merged.

Reply:
I would but one does not exist right now, seems like Kuat's change hasn't made it through yet. I don't have a `map` field though

Reply:
Pick your favorite collection and add a test for it. If no collections exists, let's tag the PR adding it to also add the test!

Reply:
Yeah there is none existing in proxyConfig. I'll make sure we add one when that change gets in